build-docker/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM alpine
+
+# build depndencies
+RUN apk update && apk add \
+  git \
+  make \
+  file \
+  autoconf \
+  automake \
+  build-base \
+  libtool \
+  db-c++ \
+  db-dev \
+  boost-system \
+  boost-program_options \
```

Comment:
This list includes a number of dependencies that are not required to compile Bitcoin Core. i.e boost-filesystem, boost-system, boost-program-options, libressl.

Reply:
I removed boost-program-options. Other docker images that the community has created contain the rest of these libraries so it seems like they are needed. It is now easy to test this out with the docker build, but I am not familiar with all the configure options.